controller.cpp

```diff
@@ -110,6 +119,7 @@ operation* controller::createOperation(operationType OpType)
 	}
 
 	return pOp;
+	delete pOp;
```

Comment:
You have to check if the pointer is not empty then delete it to avoid runtime errors. if(pOp){delete pOp;}